app/notification.go

```diff
@@ -246,9 +247,6 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 					}
 
 					if membership != nil && !membership.Following {
-						membershipsMutex.Lock()
-						participantMemberships[userID] = membership
-						membershipsMutex.Unlock()
```

Comment:
It seems that these are not needed

Reply:
I think this was done to prevent an extra store call later in this function. 
https://github.com/mattermost/mattermost-server/blob/090c4242c0ab5a5f780a14b275b0516d1633ac5a/app/notification.go#L581-L588

The loop there loops over `followers` so the profiles here definitely wont be required there.